### PR TITLE
Add type hints for tf models batch 1

### PR DESCRIPTION
**Pull request by nablabits**
# What does this PR do?

Addresses some of the type hints for tf models in https://github.com/huggingface/transformers/issues/16059:

1. `TFBlipTextModel`
2. `DPRModel` family:
	1. `TFDPRContextEncoder`
	2. `TFDPRQuestionEncoder`
	3. `TFDPRReader`
3. `LED` family
	1. `TFLEDForConditionalGeneration`
	2. `TFLEDModel`
4. `TFLxmertForPreTraining`
5. `Marian` family
	1. `TFMarianMTModel`
	2. `TFMarianModel`
6. `Rag` family
	1. `TFRagModel`
	2. `TFRagTokenForGeneration`

## Who can review?
@Rocketknight1 please

### CODE REVIEW

File: src/transformers/models/blip/modeling_tf_blip_text.py [hunk 1]
```diff
@@ -668,22 +669,22 @@ def get_extended_attention_mask(
...
-        encoder_attention_mask=None,
-        past_key_values=None,
-        use_cache=None,
-        output_attentions=None,
-        output_hidden_states=None,
-        return_dict=None,
-        is_decoder=False,
-        training=None,
-    ):
+        input_ids: TFModelInputType | None = None,
```
Comment (nablabits) [inline src/transformers/models/blip/modeling_tf_blip_text.py:672]: I've seen a nice mix of this `TFModelInputType` and `tf.Tensor | None` in a few models, so whenever I had to fill `input_ids` I defaulted to the former as  the instructions of the issue say. If this is not the case, is there a way to tell apart when each one should be used?

File: src/transformers/models/led/modeling_tf_led.py [hunk 2]
```diff
@@ -2236,24 +2236,24 @@ def get_decoder(self):
     )
     def call(
         self,
-        input_ids=None,
-        attention_mask=None,
-        decoder_input_ids=None,
-        decoder_attention_mask=None,
-        head_mask=None,
-        decoder_head_mask=None,
-        encoder_outputs: Optional[Union[Tuple, TFLEDEncoderBaseModelOutput]] = None,
```
Comment (nablabits) [inline src/transformers/models/led/modeling_tf_led.py:2245]: I've replaced this to the `|` format for consistency. [There are other places](https://github.com/huggingface/transformers/pull/25853/files#diff-8408f21e0a00169697af1da6fe22472fda43d7e5666eefe81fe55fe417a672acR1193) where there's a mix of approaches that I didn't fix as all the type hints were present. Let me know if you want me to give them some consistency

**Comment by nablabits** [comment]
It seems that some tests failed because there was a connection error to `huggingface.co`:

image

And

image

I don't seem to have permissions to retrigger them so anyone who has can do so. Alternatively we can wait until some comment needs to be addressed (pretty likely)

File: src/transformers/models/led/modeling_tf_led.py [hunk 3]
```diff
@@ -1571,7 +1571,7 @@ class TFLEDSeq2SeqLMOutput(ModelOutput):
             - 0 for tokens that are **masked**.
 
             [What are attention masks?](../glossary#attention-mask)
-        decoder_input_ids (`tf.LongTensor` of shape `(batch_size, target_sequence_length)`, *optional*):
+        decoder_input_ids (`tf.Tensor` of shape `(batch_size, target_sequence_length)`, *optional*):
```
Comment (Rocketknight1) [inline src/transformers/models/led/modeling_tf_led.py:1574]: Great catch!

File: src/transformers/models/led/modeling_tf_led.py (hunk 2 above)
Comment (Rocketknight1) [reply src/transformers/models/led/modeling_tf_led.py:2245]: I think the | format is preferred! The main reason is that it looks nicer in the automatically generated documentation

**Review by Rocketknight1 — APPROVED** [review]
This looks great! Good catches with the copy-pasted docstrings from torch as well. Is this ready to merge, or do you want to add anything else?

File: src/transformers/models/led/modeling_tf_led.py (hunk 2 above)
Comment (nablabits) [reply src/transformers/models/led/modeling_tf_led.py:2245]: Ahh, great, I've then [added a commit](https://github.com/huggingface/transformers/pull/25853/commits/b42847326f49a8e61ab822b5a53f8a5720c4595e) to make them consistent within the method

**Comment by nablabits** [comment]
> This looks great! Good catches with the copy-pasted docstrings from torch as well. Is this ready to merge, or do you want to add anything else?

Now with the consistency tweaks I'm happy for you to merge, thanks :hugs:

**Pull request merged**